flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolTest.java

```diff
@@ -770,6 +750,48 @@ public void testFailingAllocationFailsPendingSlotRequests() throws Exception {
 		}
 	}
 
+	/**
+	 * Tests that create snapshot of slots on a {@link TaskExecutor}.
+	 */
+	@Test
+	public void testCreateSlotSnapshot() throws Exception {
+
+		try (SlotPoolImpl slotPool = new SlotPoolImpl(jobId)) {
+
+			final ArrayBlockingQueue<AllocationID> allocationIds = new ArrayBlockingQueue<>(1);
+			resourceManagerGateway.setRequestSlotConsumer(
+					slotRequest -> allocationIds.offer(slotRequest.getAllocationId()));
+
+			setupSlotPool(slotPool, resourceManagerGateway, mainThreadExecutor);
+			Scheduler scheduler = setupScheduler(slotPool, mainThreadExecutor);
+
+			final SlotRequestId slotRequestId = new SlotRequestId();
+			CompletableFuture<LogicalSlot> slotRequestFuture = allocateSlot(scheduler, slotRequestId);
+
+			final List<OfferedSlot> offeredSlots = new ArrayList<>(2);
+			final List<SlotOffer> slotOffers = new ArrayList<>(2);
```

Comment:
Please be consistent with the usage of `final` and not.

Reply:
Done